x/evm/statedb/statedb.go

```diff
@@ -423,83 +439,154 @@ func (s *StateDB) Transfer(sender, recipient common.Address, amount *big.Int) {
 
 	coins := sdk.NewCoins(sdk.NewCoin(s.evmDenom, sdkmath.NewIntFromBigIntMut(amount)))
 	senderAddr := sdk.AccAddress(sender.Bytes())
+	senderBalance := s.GetBalance(sender)
```

Comment:
Collect these only if `s.evmTracer != nil && s.evmTracer.OnBalanceChange != nil`, reduce reads to StateDB if tracer is not active

Reply:
I see that martin has made the changes for this